lib/data.js

```diff
@@ -79,6 +79,56 @@ function init(env, ctx) {
     // if there's no updates done so far, just return the full set
     if (!oldData.sgvs) { return newData; }
 
+    function nsArrayTreatments(oldArray, newArray) {
+      var result = [];
+      
+      // check for add, change
+      var l = newArray.length;
+      var m = oldArray.length;
+      for (var i = 0; i < l; i++) {
+        var no = newArray[i];
+        var found = false;
+        var founddiff = false;
+        for (var j = 0; j < m; j++) {
+          var oo = oldArray[j];
+          no._id = no._id.toString();
+          if (no._id === oo._id) {
+            found = true;
+            if (!_.isEqual(oo, no)) {
+              founddiff = true;
+            }
+            break;
+          }
+        }
+        if (founddiff) {
+          var nno = _.cloneDeep(no);
+          nno.action = 'update';
+          result.push(nno);
+        }
+        if (!found) {
+          result.push(no);
+        }
+      }
+      
+      //check for delete
+      for (j = 0; j < m; j++) {
+        oo = oldArray[j];
+        found = false;
+        for (i = 0; i < l; i++) {
+          no = newArray[i];
+          if (no._id === oo._id) {
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          result.push({ _id: oo._id, action: 'remove' });
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['oo' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1542254966/issues/source?bid=2452379&fileBranchId=3015078#l125)